domain/src/main/java/org/oppia/domain/topic/TopicListController.kt

```diff
@@ -233,26 +233,37 @@ class TopicListController @Inject constructor(
       for (storySummary in topic.storyList) {
         val storyId = storySummary.storyId
         val storyProgress = storyProgressController.retrieveStoryProgress(storyId)
-        val completedChapterCount = storyProgress.chapterProgressList.count { progress ->
-          progress.playState == ChapterPlayState.COMPLETED
+
+        var completedChapterCount = 0
```

Comment:
Prefer keeping this a value to help reason about its lifecycle & correctness:

```kotlin
val completedChapterCount = storyProgress.chapterProgressMap.values.count { playState ->
  playState == ChapterPlayState.COMPLETED
}
```

Ditto for the other ``var`` below (and for variables in general--I always suggest designing your code to avoid them since it will yield the benefits of immutability in the long run). For the case below you probably want to use [find](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/find.html).